graylog2-web-interface/src/views/components/visualizations/hooks/useExtendedChartGeneratorSettings.ts

```diff
@@ -0,0 +1,95 @@
+import type { ChartDefinition } from 'views/components/visualizations/ChartData';
+
+const useExtendedChartGeneratorSettings = ({ config, barmode, effectiveTimerange }: {
+  config: AggregationWidgetConfig,
+  barmode?: BarMode,
+  effectiveTimerange: AbsoluteTimeRange,
+}) => {
+  const unitFeatureEnabled = useFeature(UNIT_FEATURE_FLAG);
+  const widgetUnits = useWidgetUnits(config);
+  const { fieldNameToAxisNameMapper, fieldNameToAxisCountMapper, unitTypeMapper } = useMemo(() => generateMappersForYAxis({ series: config.series, units: widgetUnits }), [config.series, widgetUnits]);
+  const getExtendedChartGeneratorSettings = useCallback(({ originalName, name, values }: { originalName: string, name: string, values: Array<any> }): Partial<ChartDefinition> => {
+    if (!unitFeatureEnabled) return ({});
+    const fieldNameKey = getFieldNameFromTrace({ name, series: config.series }) ?? NO_FIELD_NAME_SERIES;
+    const yaxis = fieldNameToAxisNameMapper[fieldNameKey];
+    const curUnit = widgetUnits.getFieldUnit(fieldNameKey);
+    const shouldConvertToBaseUnit = isLayoutRequiresBaseUnit(curUnit);
+    const convertedValues = shouldConvertToBaseUnit ? convertDataToBaseUnit(values, curUnit) : values;
+    const baseUnit = shouldConvertToBaseUnit && getBaseUnit(curUnit?.unitType);
+    const unit = shouldConvertToBaseUnit ? new FieldUnit(baseUnit?.unitType, baseUnit?.abbrev) : curUnit;
+
+    return ({
+      yaxis,
+      y: convertedValues,
+      ...getHoverTemplateSettings({ unit, convertedValues, originalName }),
+    });
+  }, [config.series, fieldNameToAxisNameMapper, unitFeatureEnabled, widgetUnits]);
+  const getExtendedBarsGeneratorSettings = useCallback(({
+    originalName, name, values, idx, total, xAxisItemsLength,
+  }: { xAxisItemsLength: number, originalName: string, name: string, values: Array<any>, idx: number, total: number }):Partial<ChartDefinition> => {
+    if (!unitFeatureEnabled) return ({});
+
+    const fieldNameKey = getFieldNameFromTrace({ name, series: config.series }) ?? NO_FIELD_NAME_SERIES;
+    const { y: convertedValues, yaxis, ...hoverTemplateSettings } = getExtendedChartGeneratorSettings({ originalName, name, values });
+    const axisNumber = fieldNameToAxisCountMapper?.[fieldNameKey];
+    const totalAxis = Object.keys(unitTypeMapper).length;
+
+    const offsetSettings = getBarChartTraceOffsetSettings(barmode, {
+      yaxis,
+      totalAxis,
+      axisNumber,
+      traceIndex: idx,
+      totalTraces: total,
+      effectiveTimerange,
+      isTimeline: config.isTimeline,
+      xAxisItemsLength: xAxisItemsLength,
+    });
+
+    return ({
+      yaxis,
+      y: convertedValues,
+      ...hoverTemplateSettings,
+      ...offsetSettings,
+    });
+  }, [barmode, config.isTimeline, config.series, effectiveTimerange, fieldNameToAxisCountMapper, getExtendedChartGeneratorSettings, unitFeatureEnabled, unitTypeMapper]);
+
+  return ({
```

Comment:
Right now we are always creating and returning two functions, but in the usages of the hook we are always using only one of the functions. Therefore we should split up the hook into two, for each of the two functions returned.